azurecontainerapps.ts

```diff
@@ -26,6 +26,11 @@ export class azurecontainerapps {
                 await this.authenticateAzureContainerRegistryAsync();
             }
 
+            // If a Container Registry URL was provided, try to authenticate against it
+            if (!this.util.isNullOrEmpty(this.registryUrl)) {
+                await this.authenticateContainerRegistryAsync();
+            }
```

Comment:
Let's move this block before the `acrName` check block above because `authenticateAzureContainerRegistryAsync` is going to set the value of `registryUrl` to include the value of `acrName`, so both of these if-statements will be hit